db/db_impl.cc

```diff
@@ -242,6 +243,49 @@ DBImpl::DBImpl(const DBOptions& options, const std::string& dbname,
   // we won't drop any deletion markers until SetPreserveDeletesSequenceNumber()
   // is called by client and this seqnum is advanced.
   preserve_deletes_seqnum_.store(0);
+
+  error_handler_.reset(new ErrorHandler(immutable_db_options_, &mutex_));
+}
+
+Status DBImpl::Resume() {
+  InstrumentedMutexLock db_mutex(&mutex_);
+
+  if (!error_handler_->IsDBStopped() && !error_handler_->IsBGWorkStopped()) {
+    // Nothing to do
+    return Status::OK();
+  }
+
+  Status s = error_handler_->GetBGError();
+  if (s.severity() > Status::Severity::kHardError) {
+    ROCKS_LOG_INFO(immutable_db_options_.info_log,
+        "DB resume requested but failed due to Fatal/Unrecoverable error");
```

Comment:
So hard error is something can be recovered?

Reply:
As of now, the only hard error we can recover from is running out of space in SstFileManager during background compaction/flush. The user can increase the limit and resume the DB.
Eventually, I plan to add more cases like filesystem out of space while writing the WAL, but that requires more work such as flushing memtables etc.